File: languages/ruby/src/provider/MeanAnalysis.rb
# Importando bibliotecas
require './src/model/SimpleTableAnalysis.rb'

# Criando uma classe "MeanAnalysis" que implementa a interface "SimpleTableAnalysis"
class MeanAnalysis
    include SimpleTableAnalysis

    # Método de interface, responsável por realizar a análise na lista de "UserInfo"
    # list -> Lista de "UserInfo"
    # RETURN -> média de valores da lista
    def analysis(userInfoList)
        mean = 0

        userInfoList.each do |userInfo|
            mean += userInfo.credit
        end

        return mean/userInfoList.size
    end
end

File: Languages/kotlin/src/provider/MeanAnalysis.kt
package provider

import box.UserInfo
import model.SimpleTableAnalysis

// Criando uma classe "MeanAnalysis" que implementa a interface "SimpleTableAnalysis"
class MeanAnalysis: SimpleTableAnalysis {

    /**
     * Método de interface, responsável por realizar a análise na lista de "UserInfo"
     * userInfoList: List<UserInfo> -> Lista de "UserInfo"
     * RETURN: Double -> media aritmética de valores da análise
     */
    override fun analysis(userInfoList: List<UserInfo>): Double {
        var mean: Double = 0.0
        userInfoList.forEach {
            mean += it.credit
        }

        return mean/userInfoList.size
    }
}